test/e2e/run_networking_test.go

```diff
@@ -543,56 +545,58 @@ EXPOSE 2004-2005/tcp`, ALPINE)
 		}
 	})
 
-	It("podman run network bind to 127.0.0.1", func() {
-		slirp4netnsHelp := SystemExec("slirp4netns", []string{"--help"})
-		Expect(slirp4netnsHelp).Should(ExitCleanly())
-		networkConfiguration := "slirp4netns:outbound_addr=127.0.0.1,allow_host_loopback=true"
-		port := strconv.Itoa(GetPort())
-
-		if strings.Contains(slirp4netnsHelp.OutputToString(), "outbound-addr") {
-			ncListener := StartSystemExec("nc", []string{"-v", "-n", "-l", "-p", port})
-			session := podmanTest.Podman([]string{"run", "--network", networkConfiguration, "-dt", ALPINE, "nc", "-w", "2", "10.0.2.2", port})
-			session.WaitWithDefaultTimeout()
-			ncListener.WaitWithDefaultTimeout()
-
-			Expect(session).Should(ExitCleanly())
-			Expect(ncListener).Should(Exit(0))
-			Expect(ncListener.ErrorToString()).To(ContainSubstring("Connection from 127.0.0.1"))
-		} else {
-			session := podmanTest.Podman([]string{"run", "--network", networkConfiguration, "-dt", ALPINE, "nc", "-w", "2", "10.0.2.2", port})
-			session.WaitWithDefaultTimeout()
-			Expect(session).To(ExitWithError(125, "outbound_addr not supported"))
+	for _, local := range []bool{true, false} {
+		local := local
+		testName := "HostIP"
+		if local {
+			testName = "127.0.0.1"
 		}
-	})
-
-	It("podman run network bind to HostIP", func() {
-		// Determine our likeliest outgoing IP address
-		conn, err := net.Dial("udp", "8.8.8.8:80")
-		Expect(err).ToNot(HaveOccurred())
+		It(fmt.Sprintf("podman run network slirp4netns bind to %s", testName), func() {
+			ip := "127.0.0.1"
```

Comment:
Would be cleaner if this was all done under the 
if local {
} else {
}
block above.

Reply:
No the `net.Dial("udp", "8.8.8.8:80")` should be part of the test, if you move that outside it would be executed before any test is run and then only many minutes later used. In this time the host ip might have been changed (i.e. DHCP) breaking the test. Technically the risk still exists when doing it inside but the race is super narrow here.